src/agentscope/model/_openai_model.py

```diff
@@ -148,126 +169,197 @@ async def __call__(
             kwargs["stream_options"] = {"include_usage": True}
 
         start_datetime = datetime.now()
-        response = await self.client.chat.completions.create(**kwargs)
+
+        if structured_model:
+            if tools:
+                logger.warning(
+                    "structured_model is provided. Both 'tools' and "
+                    "'tool_choice' parameters will be overridden and "
+                    "ignored. The model will only perform structured output "
+                    "generation without calling any other tools.",
+                )
+            kwargs.pop("stream", None)
```

Comment:
Why we pop the argument "stream" here?

Reply:
Since here we use the different api `self.client.chat.completions.parse` and  `self.client.chat.completions.stream`, which do not support the parameter `stream`,